src/containerapp/azext_containerapp/containerapp_env_decorator.py

```diff
@@ -113,9 +163,25 @@ def get_argument_system_assigned(self):
 
     def get_argument_user_assigned(self):
         return self.get_param("user_assigned")
+    
+    def get_argument_certificate_identity(self):
+        return self.get_param("certificate_identity")
+    
+    def get_argument_certificate_key_vault_url(self):
+        return self.get_param("certificate_key_vault_url")
 
 
 class ContainerappEnvPreviewUpdateDecorator(ContainerAppEnvUpdateDecorator):
+    def validate_arguments(self):
+        super().validate_arguments()
+
+        # validate custom domain configuration
+        if self.get_argument_hostname():
+            if self.get_argument_certificate_file() and self.get_argument_certificate_key_vault_url():
+                raise ValidationError("Cannot use certificate file/password with certificate identity/keyvaulturl at the same time")
+            if (not self.get_argument_certificate_file()) and (not self.get_argument_certificate_key_vault_url()):
```

Comment:
why not support such scenario? I think it is valid to only update host name with existing cert info?